x-pack/filebeat/input/awss3/input.go

```diff
@@ -368,3 +373,6 @@ func getProviderFromDomain(endpoint string, ProviderOverride string) string {
 	}
 	return "unknown"
 }
+
+// boolPtr returns a pointer to b.
+func boolPtr(b bool) *bool { return &b }
```

Comment:
```suggestion
func ptr[T any](v bool) *T { return &v }
```
🤣

Reply:
I've got to start thinking more generically. 😆